commit ad0bbc8f78a632b1008f624ffb6090f86731a0e0
Author: Sage Weil <sweil@redhat.com>
Date:   Fri Feb 28 15:11:37 2020 -0600

    msg: add get_{pid,random}_nonce() helpers
    
    In cases where we normally use a pid for a nonce, fall back to a random
    value when the pid == 1 (i.e., we're in a container).  For the cases where
    we use a random value, use the helper.
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index 16c82b72e88..296494ece14 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -44,7 +44,7 @@ MgrStandby::MgrStandby(int argc, const char **argv) :
 		     cct->_conf.get_val<std::string>("ms_type"),
 		     entity_name_t::MGR(),
 		     "mgr",
-		     getpid(),
+		     Messenger::get_pid_nonce(),
 		     0)),
   objecter{g_ceph_context, client_messenger.get(), &monc, NULL, 0, 0},
   client{client_messenger.get(), &monc, &objecter},
